### PR TITLE
core: add check on routes loading infra

### PR DESCRIPTION
**Pull request by flomonster**

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java [hunk 1]
```diff
@@ -148,11 +149,11 @@ private ImmutableNetwork<DiDetector, ReservationRoute> makeRouteGraph(
 
     /** Checks that the route makes sense, reports any relevant warning or error otherwise */
     private void validateRoute(ReservationRouteImpl route, RJSRoute rjsRoute) {
-        if (route.getDetectorPath().isEmpty()) {
+        if (route.getDetectorPath().size() < 2) {
             warningRecorder.register(new Warning(String.format(
-                    "Route %s has no detector on its track ranges", rjsRoute.id
+                    "Route %s has less than two detectors on its track ranges", rjsRoute.id
```
Comment (elzinko) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java]: What about getting this message from MissingDetectorsRoute ?

File: core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java [hunk 2]
```diff
@@ -148,11 +149,11 @@ private ImmutableNetwork<DiDetector, ReservationRoute> makeRouteGraph(
 
     /** Checks that the route makes sense, reports any relevant warning or error otherwise */
     private void validateRoute(ReservationRouteImpl route, RJSRoute rjsRoute) {
-        if (route.getDetectorPath().isEmpty()) {
+        if (route.getDetectorPath().size() < 2) {
```
Comment (elzinko) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java]: what about using constant for that property that needs to be reused in unit tests ?

**Comment by elzinko** [comment]
Did you plan to update / create some unit tests ?

File: core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java (hunk 2 above)
Comment (multun) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java]: it's not a magic number, it's a basic property of routes: they have an entry and exit detector

File: core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java (hunk 1 above)
Comment (flomonster) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/reservation/ReservationInfraBuilder.java]: I changed the behavior of the recorder so we can trigger multiple errors then verify and throw all of them at the same time

**Review by eckter — APPROVED** [review]

**Pull request merged**